api/authorizers/build.js

```diff
@@ -1,47 +1,76 @@
-const { User, Build } = require('../models');
+const { User, Build, Site } = require('../models');
 const buildErrors = require('../responses/buildErrors');
+const GitHub = require('../services/GitHub');
 
-const verifyBuild = (model) => {
-  const build = model && model.Builds && model.Builds[0];
+const rejectBuild = () =>
+  Promise.reject({
+    status: 403,
+    message: buildErrors.UNABLE_TO_BUILD,
+  });
 
-  if (!build) {
-    return Promise.reject({
-      status: 403,
-      message: buildErrors.UNABLE_TO_BUILD,
-    });
-  }
+const getBranch = ({ user, site, branchName }) =>
+  GitHub.getBranch(user, site.owner, site.repository, branchName)
+  .then(branch => ({
+    branch: branch.name,
+    site: site.id,
+    commitSha: branch.commit.sha,
+  }))
+  .catch(rejectBuild);
 
-  return build.get({ plain: true });
-};
+const buildExists = user => user && user.Builds && user.Builds[0];
 
 const getBuildById = (user, params) => {
   const { buildId, siteId } = params;
 
   return User.findById(user.id, {
-    include: {
+    include: [{
       model: Build,
       where: { site: siteId, id: buildId },
-    },
+    }, {
+      model: Site,
+      where: { id: siteId },
+    }],
   })
-  .then(verifyBuild);
+  .then((model) => {
+    const build = buildExists(model);
+
+    if (build) {
+      return build;
+    }
+
+    return rejectBuild();
+  });
 };
 
 const getBuildByBranch = (user, params) => {
```

Comment:
Kind of a bigger picture thought: I think the original intent for our `authorizer`s was to just check that a user had access to a specific resource. Seems like we've been leaking a bit more than that (like fetching objects) into the authorizers. We might want to have an architectural review of the approach. I think I'll file a separate `debt` issue for that discussion/evaluation.

Reply:
Filed https://github.com/18F/federalist/issues/1497

Reply:
I agree, for me it starts to get tricky when the associations are eagerly loaded, it's not clear which resource is the one you don't have access to (or doesn't exist, in this case)

Maybe it's clearer in the long run to just perform the additional DB lookups?